### PR TITLE
feat: suport ip:port for http listener

### LINKED ISSUE #510: kubesec http should permit specifying the IP to bind to
In `kubesec http` mode one can specify the port to listen on, but not an IP, so `INADDR_ANY` is always used, resulting in the listener being externally accessible absent external controls like local firewall rules.

It would be better if one could specify `i.p.ad.dr:port`. Trying to do so currently results in a usage error, because `pkg/server/server.go` unconditionally prepends ":" to the supplied port number before passing to `http.Server`.

I've made a simple patch to pass the `port` argument as-is if it contains a colon, will issue a PR. That's the minimum and lazy way to implement this, but works for me.

### PR DESCRIPTION
**Pull request by spiarh**
- Fix port setting logic
- Deprecate usage of PORT in favor of KUBESEC_ADDR
- Update README

Resolves #510

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,6 +179,10 @@ which produces the following output:
 
 Kubesec includes a bundled HTTP server
 
+The listen address for the HTTP server can be configured by setting
+`KUBESEC_ADDR` environment variable. The value can be a single port
+or an address in the form of "ip:port" or "[ipv6]:port".
```
Comment (06kellyjac) [inline README.md:184]:
```suggestion
such as `8080` or an address in the form of `ip:port` or `[ipv6]:port`.
```

or

```suggestion
such as `"8080"` or an address in the form of `"ip:port"` or `"[ipv6]:port"`.
```

File: pkg/util/util.go [hunk 2]
```diff
@@ -47,3 +50,29 @@ func Print(format string, in interface{}, w io.Writer, fn PrintTable) error {
...
+	}
+
+	ip, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return "", err
+	}
+
+	if ip != "" {
+		if i := net.ParseIP(ip); i == nil {
+			return "", fmt.Errorf("invalid IP address: %s", ip)
```
Comment (06kellyjac) [inline pkg/util/util.go:69]:
```suggestion
			return "", fmt.Errorf("invalid IP address: %q", ip)
```

to put quotes around the string

File: pkg/util/util.go [hunk 3]
```diff
@@ -47,3 +50,29 @@ func Print(format string, in interface{}, w io.Writer, fn PrintTable) error {
...
+		}
+	}
+
+	if p, err := strconv.ParseInt(port, 10, 32); err != nil || p < 1 || p > 65535 {
+		return "", fmt.Errorf("invalid port: %s", port)
```
Comment (06kellyjac) [inline pkg/util/util.go:74]:
```suggestion
		return "", fmt.Errorf("invalid port: %q", port)
```
to put quotes around the string

**Comment by spiarh** [comment]
@06kellyjac I've applied the suggestions

**Review by 06kellyjac — APPROVED** [review]

**Pull request merged**